core/domain/user_services_test.py

```diff
@@ -307,6 +338,49 @@ def test_get_user_id_from_username(self):
         self.assertIsNone(
             user_services.get_user_id_from_username('fakeUsername'))
 
+    def test_get_user_settings_from_username_returns_user_settings(self):
+        auth_id = 'someUser'
+        username = 'username'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.set_username(user_id, username)
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        user_settings = user_services.get_user_settings_from_username(username)
+
+        self.assertEqual(
+            user_settings_model.id, user_settings.user_id)
+        self.assertEqual(
+            user_settings_model.email, user_settings.email)
```

Comment:
It would typically be better to have expected values be hardcoded where possible, rather than derived from the model.

Reply:
I am not sure how to generate an oracle for the user_id since it is randomly generated. Is there a way to hardcode user_id for the assertion checks?

Reply:
Ah, my comment is only for emails. If you can't hardcode it then that's fine, I just mean hardcode the ones that can be hardcoded.